targets/simple_switch/main.cpp

```diff
@@ -20,15 +20,81 @@
 
 /* Switch instance */
 
+#ifdef BM_HAVE_DLOPEN
+#  include <dlfcn.h>
+#endif  // BM_HAVE_DLOPEN
 #include <bm/SimpleSwitch.h>
 #include <bm/bm_runtime/bm_runtime.h>
 #include <bm/bm_sim/target_parser.h>
 
+#include <sstream>
+#include <string>
+#include <vector>
+
 #include "simple_switch.h"
 
 namespace {
 SimpleSwitch *simple_switch;
-bm::TargetParserBasic *simple_switch_parser;
+
+std::string load_modules_option = "load-modules";
+
+class SimpleSwitchParser : public bm::TargetParserBasic {
+ public:
+  SimpleSwitchParser() {
+    add_flag_option("enable-swap",
+                    "enable JSON swapping at runtime");
+#ifdef BM_ENABLE_MODULES
+    add_string_option(load_modules_option,
+                      "load the given .so files as modules");
+#endif  // BM_ENABLE_MODULES
+  }
+
+  int parse(const std::vector<std::string> &more_options,
+                    std::ostream *errstream) override {
+    int result = ::bm::TargetParserBasic::parse(more_options, errstream);
+#ifdef BM_ENABLE_MODULES
+    load_modules();
+#endif  // BM_ENABLE_MODULES
+    set_enable_swap();
+    return result;
+  }
+
+ protected:
+#ifdef BM_ENABLE_MODULES
+  int load_modules() {
+    std::string modules;
+    ReturnCode retval = get_string_option(load_modules_option, &modules);
+    if (retval == ReturnCode::OPTION_NOT_PROVIDED) {
+      return 0;
+    }
+    if (retval != ReturnCode::SUCCESS) {
+      return -1;  // Unexpected error
+    }
+    std::istringstream iss(modules);
+    std::string module;
+    while (std::getline(iss, module, ',')) {
+#  ifdef BM_HAVE_DLOPEN
+      if (!dlopen(module.c_str(), RTLD_NOW | RTLD_GLOBAL)) {
+        std::clog << "WARNING: Skipping module: " << module << ": "
```

Comment:
Just one more thing. Is it possible to print this to `errstream` (as passed to the `parse` function) to be consistent with the rest of the option parsing methods?

Reply:
Sure. Will do.